src/silverlabnwb/timings.py

```diff
@@ -0,0 +1,101 @@
+import abc
+
+import numpy as np
+import pandas as pd
+
+
+class LabViewTimings(metaclass=abc.ABCMeta):
+
+    def __init__(self, relative_times_path, roi_path, dwell_time):
+        self.dwell_time = dwell_time
+        self._read_relative_times_file(relative_times_path)
+        self._read_roi_data(roi_path)
+
+    @abc.abstractmethod
+    def _read_relative_times_file(self, file_path):
+        pass
+
+    def _read_roi_data(self, roi_path):
+        self.roi_data = pd.read_csv(roi_path, sep='\t', dtype=np.float64)
+        self.n_rois = len(self.roi_data['ROI index'])
+        # in the future, we can access more shape parameters here too, for variable size ROIs in those cases,
+        # we will need to access each individual row here separately. for now, take first row and all ROIs are same
+        # size and orientation. Also, we may need to store the angle somewhere in the future (to reconstruct the ROI
+        # in 3D, in which case the ==0 assertion may become imprecise and we would need a tolerance to compare with a
+        # double value.
+        n_y_pixels = int(self.roi_data['Y stop'][0] - self.roi_data['Y start'][0])
+        n_x_pixels = int(self.roi_data['X stop'][0] - self.roi_data['X start'][0])
+        if self.roi_data['Angle (deg)'][0] == 0:
+            self.n_lines_per_roi = n_y_pixels
+            self.n_pixels_per_line = n_x_pixels
+        else:
+            self.n_lines_per_roi = n_x_pixels
+            self.n_pixels_per_line = n_y_pixels
+        if self.n_pixels_per_line == 0 and self.n_lines_per_roi == 0:
+            # assume we are in pointing mode in this case, probably better if this were passed as an argument?
+            # this class would need to know about the Modes class then though.
```

Comment:
The problem is more that this class would need to know what the value of the mode is, rather than know about the class (if that makes sense!)